reduce amount of default logging on a failed request

jsvd opened the pull request:
fixes #106

kaisecheng reviewed (commented):
I left one comment for the variable name, otherwise LGTM

jsvd commented on lib/logstash/outputs/http.rb:
```diff
@@ -252,16 +252,22 @@ def send_event(event, attempt)
-                :backtrace => exception.backtrace,
-                :will_retry => will_retry
-    )
+    log_parameters = {
```
well spotted!

jsvd commented:
Thanks for the review @kaisecheng. Your comment made me realize two things:
* this code path wasn't actually being tested anywhere in the test suite
* the tests I added were covering the part of the code that handled the responses for requests that succeeded and not when `client.send` raised an exception

I added a new commit that fixes the variable name and also moved the tests to the right place. Now if the variable is not defined, the tests will fail.

jsvd commented on spec/outputs/http_spec.rb:
```diff
@@ -165,6 +165,7 @@ def sinatra_run_wait(app, opts)
         it "should log a failure" do
           expect(subject).to have_received(:log_failure).with(any_args)
         end
+
```
```suggestion
```

jsvd commented on spec/outputs/http_spec.rb:
```diff
@@ -196,8 +196,44 @@ def sinatra_run_wait(app, opts)
         it "should make three total requests" do
           expect(subject).to have_received(:send_event).exactly(3).times
         end
+
+      end
+    end
+    context "on exception" do
+      before :each do
```
```suggestion
      end
    end

    context "on exception" do
      before :each do
```

kaisecheng reviewed (approved):
LGTM 💯 test looks good as well.
